.github/workflows/java-ci.yml

```diff
@@ -0,0 +1,28 @@
+defaults:
+  run:
+    working-directory: ./client-samples/java/rest
+name: Java CI
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@af1da67850ed9a4cedd57bfd976089dd991e2582
```

Comment:
does this need a hash version? is there a slightly more readable version or is this how they recommend we do it?

Reply:
It gets auto generated with that hash when created with actions. The github actions docs use just v4, so I'll switch to that. https://github.com/gradle/actions/blob/main/setup-gradle/README.md